### ISSUE TITLE
Not working on Unraid version 6.8.3

### ISSUE THREAD

**Issue by CodyFraker** [wolveix/satisfactory-server#19]
This looks like more of a wine+Unraid issue rather than an issue with this container itself. I'm sure there are plenty of others having a similar issue who aren't speaking up right now. 

Docker command:
`docker run -d --name='satisfactory-server' --net='host' -e TZ="America/Los_Angeles" -e HOST_OS="Unraid" -e 'STEAMUSER'='user' -e 'STEAMPWD'='pass' -e 'STEAMCODE'='code' -e 'MAXBACKUPS'='10' -e 'UDP_PORT_7777'='7777' -e 'STEAMBETA'='false' -v '/mnt/user/appdata/satisfactory':'/config':'rw' 'wolveix/satisfactory-server' `

Instructions were followed almost to a T. Theres is one extra step I had to do which was to go to the Unraid docker repo and grab a steamcmd based game from there so I could get my auth code instead of what was mentioned inside of the instructions. Aside from that I had no issues following the instructions/guide. Good job on that.

This is the issue I have on my server currently.
`0027:err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
Could not load wine-gecko. HTML rendering will be disabled.
wine: configuration in '/root/.wine' has been updated.
002f:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.`

The container remains running with no output in the console whenever I try to even connect. When I run the container I see the Steam notification saying that "XYZ Computer is Ready for Streaming" and then it says its offline shortly after making me believe that its stuck on trying to run the game but wine has had too much to drink.

I cannot connect to this within my local network. Saw OP mention they do not have much wine experience. I have zero myself so I don't know what to do either as this is a niche OS + problem. I tried to google and check the [Unraid ](https://forums.unraid.net/topic/79530-support-ich777-gameserver-dockers/?do=findComment&comment=984279)forums with little luck.

Here is some more info:

Ran wine in the container's console:
`> wineconsole`
`Microsoft Windows 6.1.7601 (3.0)

Z:\config>
`
Seems to be good there so wine is in fact running. Here is some information I found that may or may not help with this issue. I know there are plenty of people running Unraid waiting to get this sucker chuggin' along. Please keep in mind that I have no idea if these links will be useful. Let me know if you need anything else such as commands I could run against the server. 

https://stackoverflow.com/questions/21799443/wine-error-application-tried-to-create-a-window-but-no-driver-could-be-loaded
https://superuser.com/questions/902175/run-wine-totally-headless

**Comment by wolveix** [comment]
Hey! Thanks for taking the time to write out this issue :) 

>The container remains running with no output in the console whenever I try to even connect. When I run the container I see the Steam notification saying that "XYZ Computer is Ready for Streaming" and then it says its offline shortly after making me believe that its stuck on trying to run the game but wine has had too much to drink.

This is perfectly normal behaviour. The container connects to Steam to download the game, and then disconnects once the download is done. By the sound of it, unless the log is looping, your server is functioning as expected.

**Comment by wolveix** [comment]
Just read over your Docker command. Not sure why you're setting various environment variables that aren't needed; unless this is an Unraid requirement? But I doubt it. You're also not exposing port `7777`, you're just setting it via an environment variable?

**Comment by CodyFraker** [comment]
Thank you for your quick reply 👍 

> By the sound of it, unless the log is looping, your server is functioning as expected.

The log is NOT looping, so that is a relief.

> Just read over your Docker command. Not sure why you're setting various environment variables that aren't needed; 

Unraid uses a GUI to create docker containers from scratch and if you set it all up, other unraid users should be able to copy that entire command and not have to set up the container manually. long story short its to save people some time if it worked. 

>You're also not exposing port 7777, you're just setting it via an environment variable? This would explain why you're unable to connect to it. 

After re-reading the command I posted, you're totally right. Even tho in Unraid, its telling me that the ports ARE forwarded from container to host. An earlier troubleshooting step I did try was to change the container's network from bridge to host which has solved game server related issues in the past. Here is the updated command. Notice that the port flag is actually set and its not trying to set the ports as some env variable. I've never noticed it do that before. Good eye.

`docker run -d --name='satisfactory-server' --net='bridge' -e TZ="America/Los_Angeles" -e HOST_OS="Unraid" -e 'STEAMUSER'='user' -e 'STEAMPWD'='pass' -e 'STEAMCODE'='code' -e 'MAXBACKUPS'='10' -e 'STEAMBETA'='false' -p '7777:7777/udp' -v '/mnt/user/appdata/satisfactory':'/config':'rw' 'wolveix/satisfactory-server' `

I shouldn't need to port forward if I'm just testing this locally, ...right??? I'll keep digging into this. I didn't know that the error/warning was OK or not to ignore and figured that was my issue.

**Comment by wolveix** [comment]
No worries! Apologies if my message came off as hostile at all, it's late here 😅 To your credit, I should really add a notice to the README stating that most errors in the log can be ignored as long as the container continues to run without looping.

> I shouldn't need to port forward if I'm just testing this locally,...right?

You will need to on the container in the sense of exposing the port, but no you won't need to port forward via your router since you're just playing locally.

Try running your updated command and report back how it goes!

**Comment by CodyFraker** [comment]
> No worries! Apologies if my message came off as hostile at all, it's late here 😅 

Not at all! you're fine!

> Try running your updated command and report back how it goes!

No dice. I even port forwarded anyways both the container (duh) and from the router (I'm setting this up to play with friends anyways)

I can ping my server's IP from my desktop just fine so i'm wondering why this is being so fussy..

**Comment by wolveix** [comment]
Damn. How are you trying to connect to the server in Satisfactory? Via the console? What error do you get?

**Comment by CodyFraker** [comment]
> Damn. How are you trying to connect to the server in Satisfactory? Via the console? What error do you get?

Yes, through the console. `Open xxx.xxx.x.xxx` Just even tried my public facing IP and it didn't work.

Here is the full log when I start the container after it authenticates my account details.

```

Success! App '526870' already up to date.
Starting periodic command scheduler cron
...done.
0012:err:ole:marshal_object couldn't get IPSFactory buffer for interface {00000131-0000-0000-c000-000000000046}
0012:err:ole:marshal_object couldn't get IPSFactory buffer for interface {6d5140c1-7436-11ce-8034-00aa006009fa}
0012:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hres=0x80004002
0012:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, 80004002
0012:err:ole:get_local_server_stream Failed: 80004002
0014:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0014:err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
0014:err:ole:apartment_createwindowifneeded CreateWindow failed with error 87
0014:err:ole:apartment_createwindowifneeded CreateWindow failed with error 0
0014:err:ole:marshal_object couldn't get IPSFactory buffer for interface {00000131-0000-0000-c000-000000000046}
0014:err:ole:apartment_createwindowifneeded CreateWindow failed with error 14007
0014:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hres=0x800736b7
0014:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, 800736b7
0014:err:ole:get_local_server_stream Failed: 800736b7
000b:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
000b:err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
0010:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0010:err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
Could not load wine-gecko. HTML rendering will be disabled.
0027:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0027:err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
Could not load wine-gecko. HTML rendering will be disabled.
wine: configuration in '/root/.wine' has been updated.
002f:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
002f:err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
```

Maybe there is something in there more important than the display errors.


For the error within the game itself, its telling me the following:

`"UNetConnection::TIck: Connection TIMED OUT. Closing Connection.. Elapsed: 20.01, Real: 20.01, Good: 20.01, DriverTime: 20.02, Threshold: 20.00, [UNetConnection] RemoteAddr: [REDACTED], Name: IpConnection_2123xxxx, Driver: PendingNetDriver EOSNetDriver_21479blah, IsServer: NO, PC: NULL, OWNER: NULL, UniqueId: INVALID`

**Comment by wolveix** [comment]
Okay, so it is the standard connection issue. So the issue is either on your client (firewall-related. Unlikely, but I can provision a temporary server that you can try to connect to tomorrow), or on your server. Is there any way in Unraid to view open UDP ports? Or could you use something like `nmap` from your client to see whether you can reach the ip:port?

**Comment by CodyFraker** [comment]
I will keep debugging then I suppose. Feel free to send me your Discord tag and Ill add you if you wanted to see and verify everything for yourself. Doesn't have to be tonight or even the next day as you said it was late there.

**Comment by wolveix** [comment]
Sure thing, that sounds good to me. You can reach me at `Wolveix#2211` :) Once we (hopefully) get this resolved, I'll post an update in this issue so that anyone else in a similar situation in the future may benefit from what we learn.

**Comment by CodyFraker** [comment]
Something to consider:
https://forums.unraid.net/topic/39894-help-request-vnc-docker-with-wine-blueiris/page/2/

**Comment by wolveix** [comment]
After spending a few hours in a call yesterday with @CodyFraker, we determined that it was some kind of permissions issue with the mapped network storage to the Unraid server. After unmapping the storage from the container (specifically the `/config/gamefiles` directory), it worked first try. Of course I'd like to delve deeper to understand why it wasn't working, but at least for now the server is working for @CodyFraker :)

**Comment by Ephesians28** [comment]
Sorry to necro this thread, but hopefully I can add some helpful information to anyone finding this in the future.

I experienced this same UNetConnection Tick Connection TIMED OUT error when trying to set up a dedicated server for the game Satisfactory. Here's the solution I found: 

[b] Short answer: If the server you are connecting to is on your LAN, you cannot use a public IP and port to connect, you must use your local IP. [/b]

Long answer: I set up the server on dedicated server hardware, forwarded necessary ports to that IP, and set up a free DNS to point to my public IP (so I don't have to give every player my real IP address). Before telling my friends to try connecting to my server, I wanted to test it myself. I booted up the game on my desktop PC, typed in the DNS and port, and tried to connect. That gave me the error message. I pulled my hair out for hours trying to figure out why "my server was broken". In actuality, it was not broken, I just could not use my public IP (or my DNS) to connect to my own server. I tried using my local IP (192.168.1.128), and it connected with no problems. I instructed my friends to connect using my public IP or DNS and they could connect just fine.

This may not be the same cause/solution of the error message that you are experiencing, but this was the solution for me. Hope that helps.
